### PR TITLE
Add support for Anthropic models

### PR DESCRIPTION
**Pull request by kevinmingtarja**
API References:
- https://docs.anthropic.com/en/api/messages (and their [TypeScript SDK](https://github.com/anthropics/anthropic-sdk-typescript))
- https://docs.aws.amazon.com/bedrock/latest/userguide/model-parameters-anthropic-claude-messages.html

Example code:
```javascript
import { models } from "@hypermode/functions-as";
import { JSON } from "json-as";

import {
  AnthropicMessagesModel,
  Message,
  Metadata,
  TextBlock,
  Tool,
  ToolChoiceAny,
  ToolChoiceTool,
  ToolChoiceAuto,
} from "@hypermode/models-as/models/anthropic/messages";

// This model name should match the one defined in the hypermode.json manifest file.
const modelName: string = "text-generator";

export function generateText(instruction: string, prompt: string): string {
  const model = models.getModel<AnthropicMessagesModel>(modelName);
  const input = model.createInput([new UserMessage(prompt)]);
  // In AnthropicMessagesModel, system is passed as parameter to the invoke, not as a message
  input.system = instruction;

  // optional parameters
  input.temperature = 0.9;
  input.metadata = {
    userId: "user123",
  };
  // input.stopSequences = [",", ".", "?", "!"];
  input.maxTokens = 200;
  input.topK = 50;
  input.topP = 0.9;
  input.tools = [
    {
      name: "add",
      inputSchema: `{"type":"object","properties":{"x":{"type":"integer","description":"The first integer to add"},"y":{"type":"integer","description":"The second integer to add"}},"required":["x","y"]}`,
      description: "adds two numbers x and y",
    },
    {
      name: "stock_price",
      inputSchema: `{"type":"object","properties":{"symbol":{"type":"string","description":"The stock symbol"}},"required":["symbol"]}`,
      description: "gets the stock price of a symbol",
    }
  ];
  // input.toolChoice = ToolChoiceAny;
  // input.toolChoice = ToolChoiceAuto;
  input.toolChoice = ToolChoiceTool("add");

  // Here we invoke the model with the input we created.
  const output = model.invoke(input);

  const contents: string[] = []
  // The output contents can either be of type TextBlock or ToolUseBlock.
  for (let i = 0; i < output.content.length; i++) {
    if (output.content[i].type === "text") {
      contents.push((output.content[i].text || "").trim());
    } else if (output.content[i].type === "tool_use") {
      const toolUse = output.content[i];
      const toolName = toolUse.name || "";
      const inputs = toolUse.input || ""
      contents.push(`${toolName} with inputs: ${inputs}`);
    } else {
      throw new Error(`Unexpected content type: ${(output.content[i]).type}`);
    }
  }

  return contents.join(", ")
}
```

Example hypermode.json:
```json
{
  "$schema": "https://manifest.hypermode.com/hypermode.json",
  "models": {
    "text-generator": {
      "sourceModel": "claude-3-opus-20240229",
      "host": "anthropic",
      "path": "v1/messages"
    }
  },
  "hosts": {
    "anthropic": {
      "baseUrl": "https://api.anthropic.com/",
      "headers": {
        "x-api-key": "{{API_KEY}}",
        "anthropic-version": "2023-06-01"
      }
    }
  }
}
```

### CODE REVIEW

File: src/models/anthropic/messages.ts [hunk 1]
```diff
@@ -0,0 +1,430 @@
...
+
+  constructor() {
+    super("tool");
+  }
+}
+
+
+@json
+class Properties {
+  [key: string]: any;
```
Comment (mattjohnsonpint) [inline src/models/anthropic/messages.ts:309]: Can't use `any` in AssemblyScript.  But also, I think this class and also the `InputSchema` class should be removed.  The input schema is an embedded JSON Schema document - which could have anything that JSON Schema allows.  We need to find some way to have a field be dynamic enough to handle this.  I'll explore some ideas.

Comment (mattjohnsonpint) [reply src/models/anthropic/messages.ts:309]: BTW, we have the same problem in OpenAI.  I just stubbed a nullable string for now, but that's not really sufficient.

https://github.com/hypermodeAI/models-as/blob/8c6e6994096321ae032f9001d9b6aaf350d47941/src/models/openai/chat.ts#L362-L369

Comment (mattjohnsonpint) [reply src/models/anthropic/messages.ts:309]: We'll use this, when ready:
https://github.com/JairusSW/as-json/issues/88

Comment (kevinmingtarja) [reply src/models/anthropic/messages.ts]: got it, i've stubbed it with a nullable string as well, and added a TODO.

Comment (kevinmingtarja) [reply src/models/anthropic/messages.ts]: i stubbed `ToolUseBlock.input` as well for the same reason, it's an arbitrary json in the API response.

**Review by mattjohnsonpint — APPROVED** [review]

**Pull request merged**